llama_api.py

```diff
@@ -0,0 +1,169 @@
+import asyncio
+import uuid
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.embeddings import HuggingFaceEmbeddings
+from langchain.vectorstores import FAISS
+from langchain.chains.question_answering import load_qa_chain
+from langchain.prompts import PromptTemplate
+import torch
+from transformers import pipeline, LlamaForCausalLM, LlamaTokenizer
+import os
+from dotenv import load_dotenv
+import transformers
+import uvicorn
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi import FastAPI
+from pydantic import BaseModel
+from typing import List, Dict, Any
+
+load_dotenv()
+
+# Load and split the text file
+with open("general.txt", encoding="utf8") as f:
+    raw_text = f.read()
+
+text_splitter = RecursiveCharacterTextSplitter(
+    chunk_size=1000,
+    chunk_overlap=100,
+    length_function=len,
+)
+text_chunks = text_splitter.split_text(raw_text)
+
+
+embeddings = HuggingFaceEmbeddings(model_name="sentence-transformers/all-MiniLM-L6-v2")
+vector_store = FAISS.from_texts(text_chunks, embedding=embeddings)
+vector_store.save_local("faiss_index")
+
+bnb_config = transformers.BitsAndBytesConfig(
+    load_in_4bit=True,
+    bnb_4bit_quant_type='nf4',
+    bnb_4bit_use_double_quant=True,
+    bnb_4bit_compute_dtype=torch.bfloat16
+)
+
+
+model_id = "meta-llama/Llama-3.2-11B-Vision-Instruct"
+model_config = transformers.AutoConfig.from_pretrained(model_id)
+
+model = transformers.AutoModelForCausalLM.from_pretrained(
+    model_id,
+    trust_remote_code=True,
+    config=model_config,
+    quantization_config=bnb_config,
+    cache_dir='models'
+)
+
+model.eval()
+
```

Comment:
Did you load the model and do it ?
If so please use Inference APIs available in the huggingface.
Only fewer models have that option so choose wisely cause this would be easier.
![image](https://github.com/user-attachments/assets/fc987659-eb5f-4e9b-b866-17cae2b6bc68)